network/policy/policy_windows_test.go

```diff
@@ -40,11 +42,95 @@ var _ = Describe("Windows Policies", func() {
 					}}`),
 			}
 
-			expected_policy := `{"InboundProxyPort":"15003","OutboundProxyPort":"15001","FilterTuple":{"Protocols":"6"},"UserSID":"S-1-5-32-556","InboundExceptions":{},"OutboundExceptions":{}}`
+			expectedPolicy := `{"InboundProxyPort":"15003","OutboundProxyPort":"15001","FilterTuple":{"Protocols":"6"},"UserSID":"S-1-5-32-556","InboundExceptions":{},"OutboundExceptions":{}}`
 
 			generatedPolicy, err := GetHcnL4WFPProxyPolicy(policy)
 			Expect(err).To(BeNil())
-			Expect(string(generatedPolicy.Settings)).To(Equal(expected_policy))
+			Expect(string(generatedPolicy.Settings)).To(Equal(expectedPolicy))
+		})
+	})
+
+	Describe("Test GetHcnACLPolicy", func() {
+		It("Should raise error for invalid json", func() {
+			policy := Policy{
+				Type: ACLPolicy,
+				Data: []byte(`invalid json`),
+			}
+
+			_, err := GetHcnACLPolicy(policy)
+			Expect(err).NotTo(BeNil())
+		})
+
+		It("Should marshall the ACL policy correctly", func() {
+			policy := Policy{
+				Type: ACLPolicy,
+				Data: []byte(`{
+					"Type": "ACL",
+					"Protocols": "TCP",
+					"Direction": "In",
+					"Action": "Allow"
+					}`),
+			}
+			expectedPolicy := `{"Protocols":"TCP","Action":"Allow","Direction":"In"}`
+
+			generatedPolicy, err := GetHcnACLPolicy(policy)
+			Expect(err).To(BeNil())
+			Expect(string(generatedPolicy.Settings)).To(Equal(expectedPolicy))
+		})
+	})
+
+	Describe("Test GetHcnOutBoundNATPolicy", func() {
+		It("Should raise error for invalid json", func() {
+			policy := Policy{
+				Type: OutBoundNatPolicy,
+				Data: []byte(`invalid json`),
+			}
+
+			_, err := GetHcnOutBoundNATPolicy(policy, nil)
+			Expect(err).NotTo(BeNil())
+		})
+
+		It("Should marshall the OutBoundNAT policy correctly", func() {
+			policy := Policy{
+				Type: OutBoundNatPolicy,
+				Data: []byte(`{
+					"Type": "OutBoundNAT",
+					"ExceptionList": ["10.240.0.0/16","10.0.0.0/8"]
+					}`),
+			}
+			expectedPolicy := `{"Exceptions":["10.240.0.0/16","10.0.0.0/8"]}`
+
+			generatedPolicy, err := GetHcnOutBoundNATPolicy(policy, nil)
+			Expect(err).To(BeNil())
+			Expect(string(generatedPolicy.Settings)).To(Equal(expectedPolicy))
+		})
+	})
+
+	Describe("Test GetHcnRoutePolicy", func() {
```

Comment:
could we add tests to ensure nothing panics/we get expected output for each of these tests if we pass in empty json into Data or any of the fields which accept json